Sources/SourceKitLSP/Clang/ClangLanguageServer.swift

```diff
@@ -203,6 +224,17 @@ actor ClangLanguageServerShim: ToolchainLanguageServer, MessageHandler {
 
     process.standardOutput = clangdToUs
     process.standardInput = usToClangd
+    let logForwarder = ClangdStderrLogForwarder()
+    let stderrHandler = Pipe()
+    stderrHandler.fileHandleForReading.readabilityHandler = { fileHandle in
+      let newData = fileHandle.readData(ofLength: 1)
+      if newData.count == 0 {
+        stderrHandler.fileHandleForReading.readabilityHandler = nil
```

Comment:
Doesn't this cause a cyclic reference?

Reply:
Once the end of the `stderr` pipe is reached (indicated by `newData.count == 0`), we remove the `readabilityHandler` from the `stderrHandler` again, which should break the cycle.